pkg/inventory/inventory-info.go

```diff
@@ -29,5 +29,7 @@ type Info interface {
 	// if the Id is necessary and how to use it for pruning objects.
 	ID() string
 
-	Strategy() Strategy
+	// InitialInventory returns an empty initialized inventory object.
```

Comment:
I was never a big fan of this Info type. It seems confusing and overloaded.

In Config Sync, we tend to use a -Ref suffix for GKNN references, sometimes GVKNN. But I guess `inventory.Ref` is probably a worse name than `inventory.Info`, and `inventory.Reference` isn't much better. Maybe `inventory.Meta` or `inventory.Identifier` or `inventory.ID`? I guess ID would overlap with the string ID. So maybe Meta is better. Or we just leave Info, tho that's super generic.

Having the Info object also function as a factory is a bit surprising tho.

Would it make sense to have an `inventory.Factory` and separate `inventory.Meta`?

Reply:
Do we even need the name & namespace? 
Can the inventory client handle converting from ID to NN?
Do the callers actually need to know the NN or can the inventory client just handle everything using the ID?

Reply:
Did some refactoring to remove the need for Name, but we still need Namespace